### ISSUE TITLE
Fix the Message Template that appears on the CAMS Desktop issue form - currently it displays twice when resolving an issue.

### ISSUE THREAD

**Issue by richardcesdm** [exeGesIS-SDM/CAMS-Desktop#57]
This can cause confusion if users decide to edit the first message prompt, as when the second one loads it will display the original value.
For example, when resolving an issue in the CAMS_Dev_master database using the standard config values first this message template prompt will display:
![image](https://user-images.githubusercontent.com/24268198/120520739-1059d700-c3cc-11eb-82fa-0db71e7f3702.png)

(note the additional edit after the standard text value) Then after selecting the 'send' button the following message template prompt will also display:

![image](https://user-images.githubusercontent.com/24268198/120520815-25cf0100-c3cc-11eb-82ae-db9e2ae4bf9b.png)

The earlier edit has been lost in this second message prompt, but as far as I can tell the second message prompt doesn't get used for anything so it isn't a huge issue, but it can lead to confusion from the users.

I believe the reason 2 message prompts are displayed is because for certain messages CAMS will also insert a value into the WebHoldingStatusLog table. The insert needs a non-xml value vs the insert to WebHoldingStatusLog which is intended to be plain text.

I think the solution is just to prevent the second message prompt from displaying. There is no benefit to giving the officer the ability to edit the WebHoldingStatusLog value is there is no current ability to use this value, instead the WebHoldingStatusLog insert always uses the equivalent value for the message template from the WebHoldingMessageLUT.StatusDescr table.

**Comment by Timothy-D** [comment]
@richardcesdm I need some data to work on here. Can you give me something I can add to my database, please? or add to the CAMS_Dev_master database?

**Comment by richardcesdm** [comment]
within the CAMS_Dev_master database on dev-sql-1 the maint record with maintcode 00439 can be used to reproduce this problem. After the issue is set to resolved and saved, the message prompt will appear twice if the first prompt is cancelled.

**Comment by Timothy-D** [comment]
Having looked at this, I think there are two things tangled up here

**The first** looks to be a result of configuring an update message in both WebHoldingMessageLUT and MessageLUT for the same entity/field/value combination (see image below from the database)
![image](https://user-images.githubusercontent.com/25226464/128494837-489f811c-caa6-4250-8a67-9f662a0ee733.png)

The prompts to edit the message only appear if there are multiple messages configured for the same field value or the key `<add key="MaintConfig.SingleMessagePrompt" value="1"/>` is present in the config file.

In the original screenshots:
The first prompt comes from the WebHoldingMessageLUT config - the message is sent to the original web reporter of the issue, a record of this message is added to the MaintAction table. 
The second prompt comes from the MessageLUT config - the message is send to all contacts against the issue with SendUpdates =1, a record of this message is added to the MaintContactLink table.

You'll notice that the message text is different in both the screenshots, and if you manually edit them the correct edited text goes to the respective tables. I don't think this is a bug as such: I think it's just a case of having the same thing configured twice in multiple places.

If I cancel the first message, then the second message will appear, as they are independent.



**The other** is a possibility that if the first message is cancelled and multiple message templates are configured for that status, another dialogue will be shown to pick something for the Status Log. I am not sure if this is needed, so have removed this - it can be restored if needed.
This will need heavy testing with different combinations of message configuration and template options.

**Comment by richardcesdm** [comment]
Thanks Timothy, the mix up with the duplicate config across messagelut / webholdingmessagelut could explain this in the majority of cases but that is good to hear the extra entry about the status log has been removed, I think that was also a factor too.

This makes me wonder if we should actually be introducing a 'mini migration' to get clients with entries with an Entity of 'Maint' (or even all entries) in the WebHoldingMessageLUT migrated across to the MessageLUT. I am not quite sure exactly what the pros/cons for this would be but it might be easier to manage this long term if everything was managed from a single location. Might make it easier to clients to maintain by themselves too. To my knowledge the main benefit of using the MessageLUT is it can apply to all records, not just those linked to the Webholding table. I am not sure exactly how entries in the MessageLUT are configured on the view / xsl side though, or if it really would be suitable as full replacement. 

To confirm I have tested this across multiple web clients after disabling the entry in the MessageLUT and it has worked, however a better solution for those clients would likely be to migrate the WebHoldingMessageLUT config across to MessageLUT and disable the WebHoldingMessageLUT entry instead.
